partiql-eval/src/eval.rs

```diff
@@ -466,6 +486,49 @@ impl Evaluable for EvalFilter {
     }
 }
 
+/// Represents an evaluation `SelectValue` operator; `SelectValue` implements PartiQL Core's
+/// `SELECT VALUE` clause semantics. For `SelectValue` operational semantics, see section `6.1` of
+/// [PartiQL Specification — 2007](https://partiql.org/assets/PartiQL-Specification.pdf).
+#[derive(Debug)]
+pub struct EvalSelectValue {
+    pub expr: Box<dyn EvalExpr>,
+    pub input: Option<Value>,
+}
+
+impl EvalSelectValue {
+    pub fn new(expr: Box<dyn EvalExpr>) -> Self {
+        EvalSelectValue { expr, input: None }
+    }
+}
+
+impl Evaluable for EvalSelectValue {
+    fn evaluate(&mut self, ctx: &dyn EvalContext) -> Option<Value> {
+        let input_value = self.input.take().expect("Error in retrieving input value");
+
+        let ordered = &input_value.is_ordered();
+        let mut value = vec![];
+
+        for v in input_value.into_iter() {
+            let out = v.coerce_to_tuple();
+            let evaluated = self.expr.evaluate(&out, ctx);
+            value.push(evaluated);
+        }
+
+        match ordered {
+            true => Some(Value::List(Box::new(List::from(value)))),
+            false => Some(Value::Bag(Box::new(Bag::from(value)))),
+        }
+    }
+
+    fn update_input(&mut self, input: Value, _branch_num: u8) {
+        self.input = Some(input);
+    }
+}
+
+/// Represents an evaluation `Project` operator; for a given bag of input binding tuples as input
```

Comment:
Do we want to keep the `SELECT` , `SELECT VALUE`, `SELECT *` naming consistent? Currently it's modeled as `EvalProject`, `EvalSelectValue`, and `EvalProjectAll`. Wouldn't it be more consistent for them to all be:
- `EvalProject`, `EvalProjectValue`, and `EvalProjectAll`
- or `EvalSelect`, `EvalSelectValue`, and `EvalSelectAll`

Reply:
Not strong opinions, did it this way b/c thought `SELECT VALUE` is PartiQL specific but I changed the names to be more consistent in the new commit.